bin/main.rb

```diff
@@ -1,2 +1,104 @@
 #!/usr/bin/env ruby
-puts 'Hello World'
+public
+
+def valid_name(name)
+  until name[/[a-zA-Z]/]
+    puts 'Please, enter a valid name'
+    name = gets.chomp
+    valid_name(name)
+  end
+  name.upcase
+end
+
+def valid_play_again(input)
+  input.downcase!
+  if input != 'y' && input != 'n'
+    puts "Please only write 'y' or 'n'"
+    input = gets.chomp
+    valid_play_again(input)
+  else
+    true
+  end
+end
+
+def turn(player)
+  puts "#{player.name}, choose your cell!"
+  pick = gets.chomp
+  cell = pick.to_i - 1
+
+  if @board.arr[cell] == X || @board.arr[cell] == O || !pick[/\d/]
+    puts 'Invalid cell, please choose a number between 1 and 9!'
+    @board.print
+    turn(player)
+  else
+    puts "#{player.chip} on cell #{pick}!"
+    @board.arr[cell] = player.chip
+    @board.print
+  end
+end
+
+def play(player1, player2)
+  @board = Board.new
+  @board.print
+
+  until @board.full? || @board.win
+    turn(player1)
+    break if @board.full? || @board.win
+
+    turn(player2)
+  end
+end
+
+def results
+  # needs finnesing
+  if @board.win
+    puts "And the winner is: #{@board.win}!  You rock!"
+    @board.print
+  elsif @board.full?
+    puts "It's a draw! #{p1.chip}, #{p2.chip} what a wonderful battle of wits!"
+    @board.print
+  end
+end
+
+puts 'Hello! Welcome to the tic-tac-toe game!'
+puts ''
+puts 'Instructions:'
+puts '-Each player chooses a name.'
+puts '-Each player chooses a symbol (X and O are the all-time classics!)'
+puts '-The board is displayed! a 3x3 cells array! Choose by number!'
+puts '###BOARD APPEARS HERE###'
+puts '-Players take turns to choose a cell, the first one to make a line of symbols wins!'
+puts '***Have fun!***'
+puts ''
+puts ''
+
+puts 'Player one, type in your name:'
+p1_name = gets.chomp
+name1 = valid_name(p1_name)
+X = 'X'.green
```

Comment:
When I tried running your code, I got the error below:
```ruby
bin/main.rb:78:in `<main>': undefined method `green' for "X":String (NoMethodError)
```
<img width="1190" alt="Screenshot 2020-03-21 at 03 22 00" src="https://user-images.githubusercontent.com/31304943/77217487-40002480-6b23-11ea-9cb6-fdb32568b8dd.png">